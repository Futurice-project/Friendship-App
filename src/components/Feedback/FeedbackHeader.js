import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const FeedbackHeader = ({ headerText, navigateBack, onCancel }) => {
  const { viewContainer, textStyle } = styles;

  return (
    <View style={viewContainer}>
      <TouchableOpacity onPress={navigateBack}>
        <Icon
          style={{ paddingLeft: 15, fontSize: 26, textAlign: 'center' }}
          name={'ios-arrow-back'}
        />
      </TouchableOpacity>
      <Text style={textStyle}>{headerText}</Text>
      {onCancel && ( // in "report success View, you dont have the cancel button design"
        <TouchableOpacity onPress={onCancel}>
          <Text
            style={{
              fontFamily: 'NunitoSans-Regular',
              fontSize: 15,
              color: '#bbbbbb',
              textDecorationLine: 'underline',
              marginRight: 10,
            }}
          >
            Cancel
          </Text>
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = {
  viewContainer: {
    backgroundColor: '#F8F8F8',
    paddingTop: 20,
    height: 70,
    flexDirection: 'row',
    alignItems: 'center',
    shadowColor: '#d8d8d8',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    elevation: 10,
  },
  textStyle: {
    flex: 1,
    fontSize: 20,
    textAlign: 'center',
  },
};

export default FeedbackHeader;
