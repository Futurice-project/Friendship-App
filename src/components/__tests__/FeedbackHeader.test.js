import React from 'react';
import { shallow } from 'enzyme';
import FeedbackHeader from '../Feedback/FeedbackHeader';

describe('FeedbackHeader', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<FeedbackHeader />);
    expect(wrapper).toMatchSnapshot();
  });
});
