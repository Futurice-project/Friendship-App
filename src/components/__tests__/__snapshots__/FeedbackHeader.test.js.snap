// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackHeader renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedbackHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={undefined}
        >
          <Icon
            allowFontScaling={false}
            name="ios-arrow-back"
            size={12}
            style={
              Object {
                "fontSize": 26,
                "paddingLeft": 15,
                "textAlign": "center",
              }
            }
          />
        </TouchableOpacity>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        />,
        undefined,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F8F8F8",
        "elevation": 10,
        "flexDirection": "row",
        "height": 70,
        "paddingTop": 20,
        "shadowColor": "#d8d8d8",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Icon
            allowFontScaling={false}
            name="ios-arrow-back"
            size={12}
            style={
              Object {
                "fontSize": 26,
                "paddingLeft": 15,
                "textAlign": "center",
              }
            }
          />,
          "onPress": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "name": "ios-arrow-back",
            "size": 12,
            "style": Object {
              "fontSize": 26,
              "paddingLeft": 15,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": Object {
            "flex": 1,
            "fontSize": 20,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={undefined}
          >
            <Icon
              allowFontScaling={false}
              name="ios-arrow-back"
              size={12}
              style={
                Object {
                  "fontSize": 26,
                  "paddingLeft": 15,
                  "textAlign": "center",
                }
              }
            />
          </TouchableOpacity>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          />,
          undefined,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "elevation": 10,
          "flexDirection": "row",
          "height": 70,
          "paddingTop": 20,
          "shadowColor": "#d8d8d8",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Icon
              allowFontScaling={false}
              name="ios-arrow-back"
              size={12}
              style={
                Object {
                  "fontSize": 26,
                  "paddingLeft": 15,
                  "textAlign": "center",
                }
              }
            />,
            "onPress": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "name": "ios-arrow-back",
              "size": 12,
              "style": Object {
                "fontSize": 26,
                "paddingLeft": 15,
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Object {
              "flex": 1,
              "fontSize": 20,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
